app/controllers/concerns/privacy_pass_supportable.rb

```diff
@@ -0,0 +1,32 @@
+module PrivacyPassSupportable
+  extend ActiveSupport::Concern
+
+  private
+
+  def setup_privacy_pass_challenge
+    return unless privacy_pass_enabled?
+
+    tokenizer = PrivacyPassTokenizer.new
+    tokenizer.register_challenge_for_redemption(session.id)
+    challenge = tokenizer.challenge_token
+    response.set_header("WWW-Authenticate", "PrivateToken challenge=#{challenge}, token-key=#{PrivacyPassTokenizer.issuer_public_key}")
+  end
+
+  def valid_privacy_pass_redemption?
+    return false unless privacy_pass_enabled?
+    return session[:redeemed_privacy_pass] unless session[:redeemed_privacy_pass].nil?
```

Comment:
I found out this a little hard to read during testing. Would it be possible to keep the condition "positive"?

```ruby
return session[:redeemed_privacy_pass] if session.key?(:redeemed_privacy_pass)
```